src/SIS_dyn_cgrid.F90

```diff
@@ -1737,17 +1748,183 @@ subroutine basal_stress_coeff_C(G, mi, ci, sea_lev, CS)
+  real, dimension(SZI_(G),SZJ_(G)),   intent(in)  :: sea_lev !< Sea level [Z ~> m]
+  type(SIS_C_dyn_CS),                 pointer     :: CS    !< The control structure for this module
+
+  integer, parameter :: &
+           ncat_b = 100, &  ! number of bathymetry categories
+           ncat_i = 100     ! number of ice thickness categories (log-normal)
+
+  real, parameter :: &
+           max_depth = 50.0, & ! initial range of log-normal distribution (m)
+           mu_s = 0.1          ! friction coefficient
+
+  real, dimension(ncat_i) :: & ! log-normal for ice thickness
+           x_k, & ! center of thickness categories (m)
+           g_k, & ! probability density function (thickness, 1/m)
+           P_x    ! probability for each thickness category
+
+  real, dimension(ncat_b) :: & ! normal dist for bathymetry
+           y_n, & ! center of bathymetry categories (m)
+           b_n, & ! probability density function (bathymetry, 1/m)
+           P_y    ! probability for each bathymetry category
+
+  integer, dimension(ncat_b) :: tmp
+  real, dimension(0:IG%CatIce) :: hin_max   ! category limits (m)
+
+  logical, dimension (ncat_b) :: gt
+
+  real, dimension(IG%CatIce) :: vcat   ! category limits (m)
+  real, dimension(IG%CatIce) :: acat   ! category limits (m)
+
+  real :: wid_i, wid_b, mu_i, sigma_i, mu_b, m_i, v_i !  parameters for PDFs
+  real, dimension(ncat_i):: tb_tmp
+  real, dimension (SZI_(G),SZJ_(G)):: Tbt ! seabed stress factor at t point (N/m^2)
+  real :: atot, x_kmax
+  real :: cut, rho_ice, rho_water, gravit, pi, puny
+  integer :: i, ii, j, isc, iec, jsc, jec, k, n, ncat
+  real :: ci_u ! Concentration at u-points [nondim]
+  real :: ci_v ! Concentration at u-points [nondim]
+
+  isc = G%isc ; iec = G%iec ; jsc = G%jsc ; jec = G%jec
+  ncat = IG%CatIce
+  call get_SIS2_thermo_coefs(IST%ITV, rho_ice=rho_ice, rho_water=rho_water)
+  pi = 4.0*atan(1.0)
+  gravit = G%g_Earth
+
+  puny = 1.e-20
+
+! a note regarding hi_min and hin_max(0):
+! both represent a minimum ice thickness.  hin_max(0) is
+! intended to be used for particular numerical implementations
+! of category conversions in the ice thickness distribution.
+! hi_min is a more general purpose parameter, but is specifically
+! for maintaining stability in the thermodynamics.
+! hin_max(0) = 0.1 m for the delta function itd
+! hin_max(0) = 0.0 m for linear remapping
+!
+! Also note that the upper limit on the thickest category
+! is only used for the linear remapping scheme
+! and it is not a true upper limit on the thickness
+  hin_max(0)=0.0
+  do k=1,nCat
+    hin_max(k) = IG%mH_cat_bound(k)/(rho_ice*IG%kg_m2_to_H)
+  end do
+
+  Tbt=0.0
+
+  do j=jsc-1,jec+1
+    do i=isc-1,iec+1
+
+      acat(1:ncat) = IST%part_size(i,j,1:ncat)
+      atot = sum(acat)
+      vcat(1:ncat) = IST%mH_ice(i,j,1:ncat) / rho_ice
+      m_i = sum(vcat)
+
+      if (atot > 0.05 .and. m_i > 0.01 .and. sea_lev(i,j) < max_depth) then
```

Comment:
There are dimensionally inconsistent comparisons and hard-coded dimenesional constansts here. Both m_i and sea_lev have units here of [Z ~> m], while max_depth is a hard-coded number with units of [m], and 0.01 is presumbly 0.01 [m].  Please correct the units, and verify that the rescaling is correct by trying out a test case with various values Z_RESCALE_POW and R_RESCALE_POW.